pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,159 @@ type ServiceExportList struct {
 	// +listType=set
 	Items []ServiceExport `json:"items"`
 }
+
+// ServiceExportConditionType is a type of condition associated with a
+// ServiceExport. This type should be used with the ServiceExportStatus.Conditions
+// field.
+type ServiceExportConditionType string
+
+// ServiceExportConditionReason defines the set of reasons that explain why a
+// particular ServiceExport condition type has been raised.
+type ServiceExportConditionReason string
+
+// NewServiceExportCondition creates a new ServiceExport condition
+func NewServiceExportCondition(t ServiceExportConditionType, status metav1.ConditionStatus, reason ServiceExportConditionReason, msg string) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(t),
+		Status:             status,
+		Reason:             string(reason),
+		Message:            msg,
+		LastTransitionTime: metav1.Now(),
+	}
+}
+
+const (
+	// ServiceExportConditionAccepted is true when the Service Export is accepted.
+	// This does not indicate whether or not the configuration has been exported
+	// to a control plane / data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Accepted"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoService"
+	// * "InvalidServiceType"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionAccepted ServiceExportConditionType = "Accepted"
+
+	// ServiceExportReasonAccepted is used with the "Accepted" condition when the
+	// condition is True.
+	ServiceExportReasonAccepted ServiceExportConditionReason = "Accepted"
+
+	// ServiceExportReasonNoService is used with the "Accepted" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Accepted"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
```

Comment:
Before in the KEP, we treat the "Valid" true and "Conflict" false as "exported", which means it's visible outside of the local cluster.

What "Pending" and "Failed" mean may be related to "Import"? 
Does it mean that it can be exported to every imported cluster? which it's a dynamic state and very hard to track.

Reply:
>  Does it mean that it can be exported to every imported cluster? which it's a dynamic state and very hard to track.

Agreed that this semantics/state is likely impractical to express through the `Exported` condition - whether each ServiceImport is successfully created should likely be messaged through a `Ready` condition on each ServiceImport as proposed in https://github.com/kubernetes-sigs/mcs-api/pull/113

Reply:
> I'm not entirely clear on if decentralized implementations (Cilium, Submariner) will have a use for this reason.

Cilium nop as we don't really export anything technically. Submariner I am not sure, I think I grasped from @tpantelis comment that yes but I let him confirm.

Reply:
Submariner would use this condition and, in fact, already does except we currently name it `Ready`.

Reply:
Actually maybe we should also consider using "Ready" rather than exported here?

It would probably be very similar to the current thing but have a very generic "Ready" reason which can be true in addition to the exported one. That way Cilium could use that wsith the "Ready" reason and things that have some generic logic about conditions (for instance: https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus) could know about the resource condition with the Ready type. What do you all think about this idea?

Reply:
I’m fine with either one.